src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -0,0 +1,260 @@
+    """The callback version of start batch operations."""
+    cdef _BatchOperationTag batch_operation_tag = _BatchOperationTag(None, operations, None)
+    batch_operation_tag.prepare()
+
+    cdef object future = loop.create_future()
+    cdef CallbackWrapper wrapper = CallbackWrapper(future)
+    # NOTE(lidiz) Without Py_INCREF, the wrapper object will be destructed
+    # when calling "await". This is an over-optimization by Cython.
+    cpython.Py_INCREF(wrapper)
+    cdef grpc_call_error error = grpc_call_start_batch(
+        rpc_state.call,
+        batch_operation_tag.c_ops,
+        batch_operation_tag.c_nops,
+        wrapper.c_functor(), NULL)
+
+    if error != GRPC_CALL_OK:
+        raise RuntimeError("Error with callback_start_batch {}".format(error))
+
+    await future
+    cpython.Py_DECREF(wrapper)
+    cdef grpc_event c_event
+    batch_operation_tag.event(c_event)
+
+
+async def _handle_unary_unary_rpc(object method_handler, RPCState rpc_state, object loop):
+    # Receives request message
+    cdef tuple receive_ops = (
+        ReceiveMessageOperation(_EMPTY_FLAGS),
+    )
+    await callback_start_batch(rpc_state, receive_ops, loop)
+
+    # Deserializes the request message
+    cdef bytes request_raw = receive_ops[0].message()
+    cdef object request_message
+    if method_handler.request_deserializer:
+        request_message = method_handler.request_deserializer(request_raw)
+    else:
+        request_message = request_raw
+
+    # Executes application logic
+    cdef object response_message = await method_handler.unary_unary(request_message, _ServicerContextPlaceHolder())
+
+    # Serializes the response message
+    cdef bytes response_raw
+    if method_handler.response_serializer:
+        response_raw = method_handler.response_serializer(response_message)
+    else:
+        response_raw = response_message
+
+    # Sends response message
+    cdef tuple send_ops = (
+        SendStatusFromServerOperation(
+        tuple(), StatusCode.ok, b'', _EMPTY_FLAGS),
+        SendInitialMetadataOperation(tuple(), _EMPTY_FLAGS),
+        SendMessageOperation(response_raw, _EMPTY_FLAGS),
+    )
+    await callback_start_batch(rpc_state, send_ops, loop)
+
+
+async def _handle_rpc(_AioServerState server_state, RPCState rpc_state, object loop):
+    # Finds the method handler (application logic)
+    cdef object method_handler = _find_method_handler(
+        rpc_state,
+        server_state.generic_handlers
+    )
+    if method_handler is None:
+        # TODO(lidiz) return unimplemented error to client side
+        raise NotImplementedError()
+    # TODO(lidiz) extend to all 4 types of RPC
+    if method_handler.request_streaming or method_handler.response_streaming:
+        raise NotImplementedError()
+    else:
+        await _handle_unary_unary_rpc(
+            method_handler,
+            rpc_state,
+            loop
+        )
+
+
+async def _server_call_request_call(_AioServerState server_state, object loop):
```

Comment:
This doesn't need to take the entire `_AioServerState` object.

Reply:
Done.